test/mpi/coll/testlist.def

```diff
@@ -10,6 +10,8 @@ allred5 10
 allred6 4
 allred6 7
 reduce 5
+#To test intra-node topology aware trees
+reduce 5 mpiexecarg=--bind-to mpiexecarg=core
```

Comment:
Is one test sufficient for the functionality?

Reply:
Also, `-bind-to` is an mpiexec argument.  Are we supporting such arguments now?

Reply:
@yfguo You are right. This is not sufficient. I had another “HACK” patch which somehow got dropped. That will add “bind-to core” to all the tests for testing purposes. I will put it back.

Reply:
@pavanbalaji I didn’t quite get you. Intro node topology aware only makes sense and is only enabled when a rank is bound to a socket/core etc. So I was adding this “bind-to core” 
for testing purposes.

Reply:
I see.  Adding it just for testing is totally fine.